File: Documentation/5.2/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/Sorters/PutSorterServerWide.cs
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Queries.Sorting;
using Raven.Client.ServerWide.Operations.Sorters;

namespace Raven.Documentation.Samples.ClientApi.Operations.Server.Sorters
{
    public class PutSorterServerWide
    {
        public PutSorterServerWide() 
        {
            using (var store = new DocumentStore())
            {
                #region put_sorter
                // Assign the code of your custom sorter as a `string`
                string mySorterCode = "<code of custom sorter>";

                // Create the `SorterDefinition` object
                var customSorterDefinition = new SorterDefinition
                {
                    // The sorter Name must be the same as the sorter's class name in your code
                    Name = "MySorter",
                    // The Code must be compilable and include all necessary using statements
                    Code = mySorterCode
                };

                // Define the put sorters operation, pass the sorter definition
                // Note: multiple sorters can be passed, see syntax below
                var putSortersServerWideOp = new PutServerWideSortersOperation(customSorterDefinition);
                 
                // Execute the operation by passing it to Maintenance.Server.Send
                store.Maintenance.Server.Send(putSortersServerWideOp);
                #endregion
            }
        }
        
        public async Task EnableIndexAsync()
        {
            using (var store = new DocumentStore())
            {
                #region put_sorter_async
                // Assign the code of your custom sorter as a `string`
                string mySorterCode = "<code of custom sorter>";

                // Create the `SorterDefinition` object
                var customSorterDefinition = new SorterDefinition
                {
                    // The sorter Name must be the same as the sorter's class name in your code
                    Name = "MySorter",
                    // The Code must be compilable and include all necessary using statements
                    Code = mySorterCode
                };

                // Define the put sorters operation, pass the sorter definition
                // Note: multiple sorters can be passed, see syntax below
                var putSortersServerWideOp = new PutServerWideSortersOperation(customSorterDefinition);
                 
                // Execute the operation by passing it to Maintenance.Server.SendAsync
                await store.Maintenance.Server.SendAsync(putSortersServerWideOp);
                #endregion
            }
        }
        
        /*
        #region syntax_1
        public PutServerWideSortersOperation(params SorterDefinition[] sortersToAdd)
        #endregion
        */
        
        /*
        #region syntax_2
        public class SorterDefinition
        {
            public string Name { get; set; }
            public string Code { get; set; }
        }
        #endregion
        */
    }
}

File: Documentation/5.2/Samples/nodejs/ClientApi/Operations/Server/Sorters/putSorterServerWide.js
import { DocumentStore } from "ravendb";

const documentStore = new DocumentStore();

async function putSorterServerWide() {
    
        //region put_sorter
        // Create the sorter definition object
        const sorterDefinition = {
            // The sorter name must be the same as the sorter's class name in your code
            name: "MySorter",
            // The code must be compilable and include all necessary using statements (C# code)
            code: "<code of custom sorter>"
        };
        
        // Define the put sorters operation, pass the sorter definition
        const putSortersServerWideOp = new PutServerWideSortersOperation(sorterDefinition);

        // Execute the operation by passing it to maintenance.server.send
        await documentStore.maintenance.server.send(putSortersServerWideOp );
        //endregion
}

{
    //region syntax_1
    const putSortersServerWideOp = new PutServerWideSortersOperation(sortersToAdd);
    //endregion
}

{
    //region syntax_2
    // The sorter definition object 
    {
        name: string;
        code: string;
    }
    //endregion
}
